x/margin/types/proposal.go

```diff
@@ -0,0 +1,224 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	gov "github.com/cosmos/cosmos-sdk/x/gov/types/v1beta1"
+)
+
+const (
+	ProposalTypeUpdateParams  string = "UpdateParams"
+	ProposalTypeUpdatePools   string = "UpdatePools"
+	ProposalTypeWhitelist     string = "Whitelist"
+	ProposalTypeDewhitelist   string = "Dewhitelist"
+	ProposalTypeAdminCloseAll string = "AdminCloseAll"
+	ProposalTypeAdminClose    string = "AdminClose"

```

Comment:
This is how legacy proposals are managed - from SDK 47, it's managed differently, and it is managed like normal user message where executor is gov module account.
Can we use such strategy to avoid complexity?

Reply:
I don't think we should do it. It will be considered as a admin role functions.